packages/astro/src/content/runtime.ts

```diff
@@ -110,6 +134,92 @@ export function createGetEntryBySlug({
 	};
 }
 
+export function createGetDataEntryById({
+	dataCollectionToEntryMap,
+}: {
+	dataCollectionToEntryMap: CollectionToEntryMap;
+}) {
+	return async function getDataEntryById(collection: string, id: string) {
+		const lazyImport =
+			dataCollectionToEntryMap[collection]?.[/*TODO: filePathToIdMap*/ id + '.json'];
+
+		// TODO: AstroError
+		if (!lazyImport) throw new Error(`Entry ${collection} → ${id} was not found.`);
+		const entry = await lazyImport();
+
+		return {
+			id: entry.id,
+			collection: entry.collection,
+			data: entry.data,
+		};
+	};
+}
+
+export function createGetEntry({
+	getEntryImport,
+	getRenderEntryImport,
+}: {
+	getEntryImport: GetEntryImport;
+	getRenderEntryImport: GetEntryImport;
+}) {
+	return async function getEntry(
+		collectionOrLookupObject:
+			| string
+			| { collection: string; id: string }
+			| { collection: string; slug: string },
+		lookupId?: string
+	) {
```

Comment:
nit: I find this part of the code difficult to read, probably because of these inline types. And there's also the fact that this function doesn't have a return type... so while the compiler can infer what's returning, I honestly can't understand it 😆 

Any chance to avoid inline types, at least in this case?

Reply:
Refactored to explicit return types and added an inline comment for the weird naming